File: 14. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        idx = -1
        if len(strs) == 0:
            return ''
        for i in range(len(strs[0])):
            for s in strs:
                if len(s) - 1 < i:
                    return '' if idx == -1 else strs[0][:idx+1]
                if s[i] != strs[0][i]:
                    return '' if idx == -1 else strs[0][:idx+1]
            idx = i
        return strs[0]

File: leetcode/14. Longest Common Prefix.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        let v: Vec<Vec<char>> = strs.into_iter().map(|s| {
            return s.chars().collect();
        }).collect();
        
        let mut res = "".to_string();
        
        for i in (0..v[0].len()) {
            let mut c = v[0][i];
            for j in (1..v.len()) {
                if v[j].len() < i + 1 {
                    return res;
                } else if v[j][i] != c {
                    return res;
                }
            }
            res.push(c);
        }
        
        return res;
        
    }
}
